debian/update.sh

```diff
@@ -50,77 +50,51 @@ if [ ! -L "$INSTDIR/linux" ] ; then
-fi
-(cd linux; eval $make -j$NPROC modules_install INSTALL_MOD_PATH="$INSTDIR/modules")
-mv "$INSTDIR/modules/lib/modules"/* "$INSTDIR/modules"
-rm -r "$INSTDIR/modules/lib"
-rm "$INSTDIR/modules"/*/{build,source}
-
-# install CM1 dtbs
 [ -d "$INSTDIR/boot/overlays" ] || mkdir "$INSTDIR/boot/overlays"
 rm -f "$INSTDIR/boot"/*.dtb "$INSTDIR/boot/overlays"/*.dtbo
-(cd linux; eval $make -j$NPROC dtbs_install INSTALL_DTBS_PATH=/tmp/dtb.$$)
-mv /tmp/dtb.$$/*.dtb "$INSTDIR/boot"
-mv /tmp/dtb.$$/overlays/* "$INSTDIR/boot/overlays"
-rmdir /tmp/dtb.$$/overlays /tmp/dtb.$$
-
-# build CM3 kernel
-make+=7
-BUILDDIR+=7
-rm -rf "$BUILDDIR"
-mkdir "$BUILDDIR"
-(cd linux; eval $make revpi-v7_defconfig)
-(cd linux; eval $make -j$NPROC zImage modules dtbs 2>&1)
-version="$(cat "$BUILDDIR/include/config/kernel.release")"
-copy_files
-
-# build CM3 piKernelMod
-if [ -d "$PIKERNELMODDIR" ] ; then
-	cd "$PIKERNELMODDIR"
-	make compiletime.h
-	cd -
-	(cd linux; eval $make M="$PIKERNELMODDIR" modules)
-fi
 
-# install CM3 kernel
-cp "$BUILDDIR/arch/arm/boot/zImage" "$INSTDIR/boot/kernel7.img"
-
-# install CM3 modules
-if [ -d "$PIKERNELMODDIR" ] ; then
-	(cd linux; eval $make -j$NPROC modules_install INSTALL_MOD_PATH="$INSTDIR/modules" M="$PIKERNELMODDIR")
-fi
-(cd linux; eval $make -j$NPROC modules_install INSTALL_MOD_PATH="$INSTDIR/modules")
-mv "$INSTDIR/modules/lib/modules"/* "$INSTDIR/modules"
-rm -r "$INSTDIR/modules/lib"
-rm "$INSTDIR/modules"/*/{build,source}
-
-# install CM3 dtbs
-(cd linux; eval $make -j$NPROC dtbs_install INSTALL_DTBS_PATH=/tmp/dtb.$$)
-mv /tmp/dtb.$$/*.dtb "$INSTDIR/boot"
-mv /tmp/dtb.$$/overlays/* "$INSTDIR/boot/overlays"
-rmdir /tmp/dtb.$$/overlays /tmp/dtb.$$
+# CM1=6 CM3=7 CM4=7l
+kernel_versions="6 7 7l"
+
+for kernel_version in $kernel_versions; do
+    rm -rf "$BUILDDIR"
+    mkdir "$BUILDDIR"
+
+    [ ! -d "extra" ] && mkdir "extra"
+    echo "_ _ $version" > extra/uname_string
+
+    # build kernel
+    (cd linux; eval $make revpi-v${kernel_version}_defconfig)
+    (cd linux; eval $make -j$NPROC zImage modules dtbs 2>&1)
+    version="$(cat "$BUILDDIR/include/config/kernel.release")"
+    copy_files
+
+    # build piKernelMod
+    if [ -d "$PIKERNELMODDIR" ] ; then
+      cd "$PIKERNELMODDIR"
+      make compiletime.h
+      cd -
+      (cd linux; eval $make M="$PIKERNELMODDIR" modules)
+    fi
+
+    # install kernel
+    cp "$BUILDDIR/arch/arm/boot/zImage" "$INSTDIR/boot/kernel${kernel_version/6/}.img"
+
+    # install modules
+    if [ -d "$PIKERNELMODDIR" ] ; then
+      (cd linux; eval $make -j$NPROC modules_install INSTALL_MOD_PATH="$INSTDIR/modules" M="$PIKERNELMODDIR")
+    fi
+    (cd linux; eval $make -j$NPROC modules_install INSTALL_MOD_PATH="$INSTDIR/modules")
+    mv "$INSTDIR/modules/lib/modules"/* "$INSTDIR/modules"
+    rm -r "$INSTDIR/modules/lib"
+    rm "$INSTDIR/modules"/*/{build,source}
+
+    # install dtbs
```

Comment:
Actually this can be pulled out of the for loop and performed only once after running the three builds.

Reply:
Likewise, in case it's not clear,  I'm referring to the dtb installation.

Reply:
Isn't the make command scoped with the specific `builddir` or is it enough. if this task is executed once (for example with the latest `builddir`)?

Reply:
Executing once should be sufficient. (The generated dtb's should be identical for different Kconfig settings.)